vta/scripts/tune_resnet.py

```diff
@@ -184,7 +184,7 @@ def tune_tasks(
 
         # create tuner
         if tuner == "xgb" or tuner == "xgb-rank":
```

Comment:
```tuner == "xgb-rank"``` should be changed too.

Reply:
@FrozenGene 
> `tuner == "xgb-rank"` should be changed too.

Thanks for the observation !

* fixed and exposed properly all such cases.
* exposed / updated tmvc with extended options coverage.
* extended the demo lists (taken from original ```tune_relay_arm.py``` for completeness)

Reply:
@FrozenGene ,

Uptade

> @FrozenGene
> 
> > `tuner == "xgb-rank"` should be changed too.
> 
> Thanks for the observation !
> 
> * fixed and exposed properly all such cases.
> * exposed / updated tmvc with extended options coverage.
> * extended the demo lists (taken from original `tune_relay_arm.py` for completeness)

Plus:

* leaved "rank" types in the list for compatibility with xgboost < 2.0.0